### PR TITLE
ENG-2654/ENG-2690 bundles status endpoint

### PR DESCRIPTION
**Pull request by firegloves**

### CODE REVIEW

File: src/main/java/org/entando/kubernetes/controller/digitalexchange/component/EntandoBundleResourceController.java [hunk 1]
```diff
@@ -64,4 +75,35 @@
...
+            throw new EntandoValidationException("Empty list of bundles received");
+        }
+
+        List<BundlesStatusItem> invalidBundlesStatusItemList = new ArrayList<>();
+        List<URL> repoUrlList = new ArrayList<>();
+
+        for (String stringUrl : bundlesStatusQuery.getIds()) {
+            try {
+                URL url = new URL(stringUrl);
+                ValidationFunctions.validateUrlOrThrow(url, "The rceived url is empty", "The received url is not valid");
```
Comment (cecchisandrone) [inline src/main/java/org/entando/kubernetes/controller/digitalexchange/component/EntandoBundleResourceController.java]: received

File: src/main/java/org/entando/kubernetes/model/bundle/BundleStatus.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    INSTALLED("INSTALLED"),
+    INSTALLED_NOT_DEPLOYED("INSTALLED_NOT_DEPLOYED"),
+    INVALID_REPO_URL("INVALID_REPO_URL");
+
+    private final String status;
+
+    private static final Map<String, BundleStatus> statusToBundleType;
+
+    static {
+        Map<String, BundleStatus> map = new ConcurrentHashMap<>();
```
Comment (cecchisandrone) [inline src/main/java/org/entando/kubernetes/model/bundle/BundleStatus.java:24]: Is `ConcurrentHashMap` really needed here?

File: src/main/java/org/entando/kubernetes/model/bundle/status/BundlesStatusItem.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Accessors(chain = true)
+@JsonInclude
+public class BundlesStatusItem {
+
+    private String id;
+    private BundleStatus status;
+    private String installedVersion;
```
Comment (cecchisandrone) [inline src/main/java/org/entando/kubernetes/model/bundle/status/BundlesStatusItem.java:19]: As this is a DTO returned by controller, isn't better to use snake case for the json property (`installed_version`)?

File: src/main/java/org/entando/kubernetes/model/bundle/status/BundlesStatusItem.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
+import org.entando.kubernetes.model.bundle.BundleStatus;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Accessors(chain = true)
+@JsonInclude
```
Comment (cecchisandrone) [inline src/main/java/org/entando/kubernetes/model/bundle/status/BundlesStatusItem.java:14]: Is this really needed?

**Comment by user-54dbb57a** [comment]
@firegloves 
rebased again to the latest develop to move to version 1.1 of the pipelines

File: src/main/java/org/entando/kubernetes/model/bundle/status/BundlesStatusItem.java (hunk 3 above)
Comment (firegloves) [reply src/main/java/org/entando/kubernetes/model/bundle/status/BundlesStatusItem.java:19]: yes theoretically, but we don't use this approach anywhere. we use camelcase for DTO to return too

**Review by user-54dbb57a — APPROVED** [review]

**Pull request merged**